grpc/pb-marshalling.go

```diff
@@ -178,6 +179,8 @@ func PBToValidationRecord(in *corepb.ValidationRecord) (record core.ValidationRe
 		AddressUsed:       addrUsed,
 		URL:               in.Url,
 		AddressesTried:    addrsTried,
+		// TODO(#7140): Add ResolverAddress only after the core.proto change is
+		// deployed.
```

Comment:
You can add this now, as long as the field being present isn't _required_.

Reply:
Right. If you add `ResolverAddress: in.ResolverAddress` you'll get one of two results:
1) The ResolverAddress field of the protobuf was set (because the ValidationRecord came from an updated VA), and so we copy it into the core struct.
2) The ResolverAddress field of the protobuf was not set (because the ValidationRecord came from an old VA), so it has its zero value, and so we also set the field in the core struct to its zero value as well.

Throughout this PR, we should be reading from the field. We just can't be requiring that the value read is not the zero value.

Reply:
Thank you for breaking that down.